src/main/java/com/minelittlepony/api/pony/network/MsgPonyData.java

```diff
@@ -41,6 +45,19 @@ private void initTriggerPixels(Map<String, TriggerPixelType<?>> map) {
     }
 
     public MsgPonyData(PacketByteBuf buffer) {
+        short data = buffer.readShort();
+        if (data != API_IDENTIFIER || buffer.readByte() != API_VERSION) {
```

Comment:
Won't this cause a problem with the offset if you return without consuming the rest of the buffer?